commit 517846c85dfc48aa231e28e95e8f90a6d8a8efde
Author: Florian Weimer <fweimer@redhat.com>
Date:   Wed Mar 12 11:29:10 2025 +0100

    Makefile: Clean up pthread_atfork integration
    
    Do not add the pthread_atfork routine again in nptl/Makefile,
    instead rely on sysdeps/pthread/Makefile for the integration
    (as this is the directory that contains the source file).
    
    In sysdeps/pthread/Makefile, add to static-only-routines.
    
    Reviewed-by: Joseph Myers <josmyers@redhat.com>

diff --git a/nptl/Makefile b/nptl/Makefile
index 455703bbd763d516..d03846c2e04afa2f 100644
--- a/nptl/Makefile
+++ b/nptl/Makefile
@@ -58,7 +58,6 @@ routines = \
   old_pthread_cond_signal \
   old_pthread_cond_timedwait \
   old_pthread_cond_wait \
-  pthread_atfork \
   pthread_attr_copy \
   pthread_attr_destroy \
   pthread_attr_extension \
@@ -204,7 +203,6 @@ routines = \
   vars \
   # routines
 
-static-only-routines = pthread_atfork
 libpthread-routines = libpthread-compat
 libpthread-shared-only-routines = libpthread-compat
 
diff --git a/sysdeps/pthread/Makefile b/sysdeps/pthread/Makefile
index 58c33a8e49d517ad..fa70a33e25f23230 100644
--- a/sysdeps/pthread/Makefile
+++ b/sysdeps/pthread/Makefile
@@ -145,7 +145,7 @@ tests-time64 := \
   tst-sem5-time64 \
   tst-thrd-sleep-time64 \
 
-static-only-routines = pthread_atfork
+static-only-routines += pthread_atfork
 
 # Files which must not be linked with libpthread.
 tests-nolibpthread += tst-unload
